Jenkinsfile

```diff
@@ -41,95 +34,90 @@ pipeline {
     stages {
         stage('Checkout') {
             agent any
+            environment{ STAGE = "Checkout" }
             steps {
                 deleteDir()
                 checkout scm
                 stash name: 'app'
-                sh "$SBT version"
+                sh "sbt version"
                 script {
                     version = '1.0.' + env.BUILD_NUMBER
                     currentBuild.displayName = version
-                    STAGE = "Checkout"
                 }
             }
         }
 
         stage('Build') {
             agent any
             steps{
-                sh "$SBT clean compile"
+                sh "sbt clean compile"
+            }
+            post {
+                success {
+                    colourText("info","Stage: ${env.STAGE} successful!")
+                }
+                failure {
+                    colourText("warn","Stage: ${env.STAGE} failed!")
+                }
             }
         }
 
         stage('Test'){
             agent any
+            environment{ STAGE = "Test"  }
             steps {
                 colourText("info", "Building ${env.BUILD_ID} on ${env.JENKINS_URL} from branch ${env.BRANCH_NAME}")
 
-                sh """
-                    $SBT clean compile "project $MODULE_NAME" coverage test coverageReport coverageAggregate
-                """
+                sh 'sbt coverage test coverageReport coverageAggregate'
             }
             post {
-                always {
-                    script {
-                        STAGE = "Test"
-                    }
-                }
                 success {
-                    colourText("info","Tests successful!")
+                    junit '**/target/test-reports/*.xml'
+                    step([$class: 'CoberturaPublisher', coberturaReportFile: '**/target/coverage-report/*.xml'])
+                    colourText("info","Stage: ${env.STAGE} successful!")
                 }
                 failure {
-                    colourText("warn","Failure during tests!")
+                    colourText("warn","Stage: ${env.STAGE} failed!")
                 }
             }
         }
 
         stage('Static Analysis') {
             agent any
+            environment{ STAGE = "Static Analysis" }
             steps {
                 parallel (
                     "Scalastyle" : {
```

Comment:
I assume IT tests will live here as well - although I don't think theres any for this project (which is another task in itself)

Reply:
I avoid that. If you run it after the faster unit and static analysis, you save CPU resource in the event one of them fails.